presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -304,9 +306,17 @@ public boolean schemaExists(Session session, CatalogSchemaName schema)
         ConnectorSession connectorSession = session.toConnectorSession(connectorId);
         List<ConnectorTableLayoutResult> layouts = metadata.getTableLayouts(connectorSession, connectorTable, constraint, desiredColumns);
 
-        return layouts.stream()
+        // return an empty list if it contains nothing before filtered
+        if (layouts.isEmpty()) {
+            return ImmutableList.of();
+        }
+
+        List<TableLayoutResult> resultLayouts = layouts.stream()
                 .map(layout -> new TableLayoutResult(fromConnectorLayout(connectorId, transaction, layout.getTableLayout()), layout.getUnenforcedConstraint()))
+                .filter(layoutHasAllDesiredColumn(desiredColumns))
                 .collect(toImmutableList());
+        checkState(!resultLayouts.isEmpty(), "No usable layouts");
```

Comment:
`checkState` throws `IllegalStateException` which *"Signals that a method has been invoked at an illegal or inappropriate time."* That is the the case here, so this isn't the right exception.

Do it like this instead:
```java
checkCondition(!resultLayouts.isEmpty(), NO_TABLE_LAYOUTS, "No usable layouts for table");
```
Add this as a new internal error.